cytominer_database/ingest_incl_parquet.py

```diff
@@ -0,0 +1,203 @@
+import pandas as pd
+import backports.tempfile
+import sqlalchemy.exc
+from sqlalchemy import create_engine
+import pyarrow
+import pyarrow.parquet as pq
+import pyarrow.csv
+import numpy as np
+import collections
+import numpy as np
+import cytominer_database
+import cytominer_database.utils
+import cytominer_database.write
+import cytominer_database.tableSchema
+
+def seed(source, output_path, config_path, skip_image_prefix=True, directories=None):
+    """
+    Main function. Loads configuration. Opens ParquetWriter.
+    Calls writer function. Closes ParquetWriter.
+    :source: path to parent directory containing subdirectories, e.g. "path/plate_a/"
+    :output_path: path of destination folder
+    :config_path: full path of configuration file, e.g. path/to/plate_a/ingest_config.ini
+    :skip_image_prefix: Boolean value specifying if the column headers of
+     the image.csv files should be prefixed with the table name ("Image").
+    :directories: Pass subdirectories path list instead of generating a
+        path list from the source path. Added to test special cases. Can be removed.
+
+    """
+    config = cytominer_database.utils.read_config(config_path)
+    engine = config["ingestion_engine"]["engine"]
+
+    # get dictionary that contains [name]["writer"], [name]["schema"], [name]["pandas_dataframe"]
+    writers_dict = cytominer_database.tableSchema.open_writers(source, output_path, config, skip_image_prefix)
+    # lists the subdirectories that contain CSV files
+    if not directories:
+        directories = sorted(
+            list(cytominer_database.utils.find_directories(source))
+        ) 
+    # ----------------------------- iterate over subfolders in source folder------------------------------------
+    for directory in directories:
+        # ....................... get input .csv file paths ......................
+        try:
+            compartments, image = cytominer_database.utils.validate_csv_set(config, directory)
+        except IOError as e:
+            click.echo(e)
+            continue
+
+        identifier = checksum(image)
+        csv_locations = [image] + compartments
+        # ----------------------------------- iterate over .csv's ---------------------------------------
+        for input_path in csv_locations:
+                table_name = cytominer_database.utils.get_name(input_path)
+                dataframe = cytominer_database.load.get_and_modify_df(input_path, identifier, skip_image_prefix)
+                cytominer_database.utils.type_convert_dataframe(dataframe, config, engine)
+                cytominer_database.write.write_to_disk(dataframe, table_name, output_path, engine, writers_dict)
+    # --------------------------------------- close writers ---------------------------------------------
+    close_writers(writers_dict, engine)
+# --------------------------------------------- end ---------------------------------------------------
+
+
+def close_writers(writers_dict, engine):
```

Comment:
Is this function ever used if `engine != "Parquet"`. It looks like the function doesn't do anything if `engine != "Parquet"`